test/model/narrow_test.dart

```diff
@@ -149,4 +149,17 @@ void main() {
       check(narrow123.containsMessage(dm(user3, [user1, user2]))).isTrue();
     });
   });
+
+  group('MentionsNarrow', () {
+    test('containsMessage', () {
+      const narrow = MentionsNarrow();
+
+      check(narrow.containsMessage(eg.streamMessage(
+        flags: []))).isFalse();
+      check(narrow.containsMessage(eg.streamMessage(
+        flags:[MessageFlag.mentioned]))).isTrue();
+      check(narrow.containsMessage(eg.streamMessage(
+        flags: [MessageFlag.wildcardMentioned]))).isTrue();
```

Comment:
nit: this is a bit hard to read because the `isFalse` and `isTrue`, which are the most critical bits of information, get visually buried.

This reformatting helps make them stick out:

```suggestion
      check(narrow.containsMessage(
        eg.streamMessage(flags: []))).isFalse();
      check(narrow.containsMessage(
        eg.streamMessage(flags: [MessageFlag.mentioned]))).isTrue();
      check(narrow.containsMessage(
        eg.streamMessage(flags: [MessageFlag.wildcardMentioned]))).isTrue();
```

Reply:
Stepping back, though, the main purpose of a test is to keep the code from getting accidentally broken by future changes. Here, I think the biggest risk for accidental breakage is when we add another mentions-related flag, that we might forget to handle it in MentionsNarrow.containsMessage. This test as written won't help in that case — it won't get broken by that.

Probably the most effective thing to do is to make the method's implementation use an exhaustive switch rather than a pair of equality checks. Then it's fine to still have this test, but there's no longer so much relying on it.